src/game/Tactical/Overhead.cc

```diff
@@ -1523,37 +1524,26 @@ BOOLEAN HandleGotoNewGridNo(SOLDIERTYPE* pSoldier, BOOLEAN* pfKeepMoving, BOOLEA
 					bPosOfMask = NO_SLOT;
 				}
 
-				const ExplosiveModel* pExplosive = nullptr;
-				if (!AM_A_ROBOT(pSoldier))
+				const SmokeEffectModel* smokeEffect = nullptr;
+				auto smokeEffectID = GetSmokeEffectOnTile(pSoldier->sGridNo, pSoldier->bLevel);
```

Comment:
Is there any risk of applying consequences several times in a turn this way? Checking the tile for presence vs `fHitByGasFlags`, which seems to be a marker specifically for such cases.

Reply:
True I've added `IsSoldierAlreadyAffectedBySmokeEffect` to reflect the correct behavior.